src/snowflake/connector/pandas_tools.py

```diff
@@ -225,6 +228,12 @@ def write_pandas(
         parquet_columns = "$1:" + ",$1:".join(f'"{c}"' for c in df.columns)
     else:
         parquet_columns = "$1:" + ",$1:".join(df.columns)
+    parquet_columns = ",".join(
+        [
+            "$1:"+x if x not in column_formats else column_formats[x].substitute(col=x)
+            for x in df.columns
+        ]
```

Comment:
```suggestion
        "$1:" + x if x not in column_formats else column_formats[x].substitute(col=x)
        for x in df.columns
```
I think list comprehension can be skipped and you can use generator.

Reply:
fixed